File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums: List[int]) -> int:
        dc=defaultdict(list)

        def totalDigits(x):
            total=0
            while x>0:
                digit = x%10
                total+=digit
                x//=10
            return total

        largest_sum=-1

        for num in nums:
            total=totalDigits(num)
            if total in dc:
                last=dc[total][-1]
                if num>=last:
                    dc[total].append(num)
                
                if last+num>largest_sum:
                    largest_sum=last+num
                    
            else:
                dc[total].append(num)


        return largest_sum

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.ts
function maximumSum(nums: number[]): number {
    let storage = new Map<number, number>()
    let res = -1

    for (let i = 0; i < nums.length; i++) {
        let str_num = nums[i].toString();
        let total = 0
        for (let digit of str_num) {
            total += Number(digit)
        }
        if (!storage.has(total)) {
            storage.set(total, nums[i])
        }
        else {
            let compared = storage.get(total)
            res = Math.max(res, compared + nums[i])
            if (nums[i] > compared) {
                storage.set(total, nums[i])
            }
        }

    }

    return res
};